File: baekjoon/greedy/16112.py
# nodejs로 무슨 짓을 해도 안 풀려서 파이썬으로 풀었다. 근데 풀 필요가 없었다.
quests, maximumArcanes = map(int, input().split())
gainingExperiences = list(map(int, input().split()))
gainingExperiences.sort()

sum = 0

for i in range(0, len(gainingExperiences)):
    if(i > maximumArcanes) :
        sum += gainingExperiences[i] * maximumArcanes
    elif(i - maximumArcanes <= 0):
        sum += gainingExperiences[i] * (maximumArcanes - abs(i - maximumArcanes))

print(sum)

File: baekjoon/greedy/16112.js
// S2
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
 });
 
let input = [];
 
rl.on('line', function (line) {
    input.push(line)
 })
.on('close', function () {
    const [quests, maximumArcanes] = input[0].split(' ').map(v => parseInt(v));
    const gainingExperiences = input[1].split(' ').map(v => parseInt(v));
    gainingExperiences.sort((a, b) => a - b);

    let sum = 0;
    let count = 0;

    gainingExperiences.forEach((exp, i) => {
        if(i > maximumArcanes) {
            sum += exp * maximumArcanes;
        }
        else if(i - maximumArcanes <= 0) {
            sum += exp * (maximumArcanes - Math.abs((i - maximumArcanes)));
        }
    });
    
    console.log(sum);
});
/**
 * nodejs로는 무슨 짓을 해도 채점 시작하자마자 바로 틀렸다고 떠서 파이썬으로 푼 문제.
 */